### ISSUE TITLE
Warn about using `bool::then_some` followed by `unwrap_or`, suggest `if`/`else` instead

### ISSUE THREAD

**Issue by joshtriplett** [rust-lang/rust-clippy#9100]
### What it does

Detects the combination of `then_some` followed by `unwrap_or` in place of a more clear `if`/`else`.

### Lint Name

bool_if_else

### Category

style

### Advantage

- Clarity at a glance

### Example

```rust
let mode = executable.then_some(IsExecutable).unwrap_or(IsNotExecutable);
```

Could be written as:

```rust
let mode = if executable { IsExecutable } else { IsNotExecutable };
```

**Comment by joshtriplett** [comment]
This should also apply with any combination of `then`/`then_some` and `unwrap_or`/`unwrap_or_else`.

**Comment by arieluy** [comment]
My commit adds the lint for `then_some`/`unwrap_or`, more commits needed to address other cases. I'll get to them at some point, but anyone can feel free to take them before me.
